src/cosalib/qemuvariants.py

```diff
@@ -205,9 +207,15 @@ def mutate_image(self, callback=None):
         cmd.extend([work_img])
         run_verbose(cmd)
 
-        if callback:
+        img_info = image_info(work_img)
+        if self.image_format != img_info.get("format"):
+            raise ImageError((f"{work_img} format mismatch"
+                              f" expected: '{self.image_format}'"
+                              f" found: '{img_info.get('format')}'"))
```

Comment:
Ouhh, that's a nice check! Almost worth a separate commit, but up to you!

Aside, re. 

> Finally, a format check was added to ensure that the format is
appropriately changed. In some cases, 'qemu-img convert' will produce
'raw' images instead of compliant VHD's.

did you see https://github.com/coreos/coreos-assembler/issues/1091#issuecomment-582491137? Did you find out more?

Reply:
Yes, I did :) 

However, the file magic was actually missing :)